src/accelerate/commands/launch.py

```diff
@@ -151,6 +151,44 @@ def multi_gpu_launcher(args):
         raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
 
 
+def deepspeed_launcher(args):
+    # TODO: need to complete this
+
+    cmd = ["deepspeed"]
+    if args.num_machines > 1:
+        cmd.extend(
+            [
+                "--num_gpus",
+                str(args.num_processes // args.num_machines),
+                "--num_nodes",
+                str(args.num_machines),
+                "--node_rank",
+                str(args.machine_rank),
+                "--master_addr",
+                args.main_process_ip,
+                "--master_port",
+                str(args.main_process_port),
+            ]
+        )
+    else:
+        cmd.extend(["--num_gpus", str(args.num_processes)])
+
+    cmd.append(args.training_script)
+    cmd.extend(args.training_script_args)
+
+    current_env = os.environ.copy()
+    current_env["USE_FP16"] = str(args.fp16)
+    current_env["ZERO_STAGE"] = int(args.zero_stage)
+    current_env["GRADIENT_ACCUMULATION_STEPS"] = int(args.gradient_accumulation_steps)
+    current_env["IS_TRAIN_BATCH_MIN"] = bool(args.is_train_batch_min)
```

Comment:
Are those all expected by deepspeed?

Reply:
yes in `deepspeed.initialize()` as `config_params`